lib/Gateway.js

```diff
@@ -1361,9 +1366,13 @@ Gateway.prototype.discoverValue = function (node, vId) {
 
         Object.assign(cfg.discovery_payload, sensor.props || {})
 
-        if (valueId.units) {
+        // Discovered valueId.unit also exists. therefore added both unit and units to be
+        // checked, but only one used
+        if (valueId.unit) {
           // https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/scales.json
-          cfg.discovery_payload.unit_of_measurement = valueId.units
+          cfg.discovery_payload.unit_of_measurement = valueId.unit
+        } else if (valueId.units) {
+          cfg.discovery_payload.unit_of_measurement = valueId.unit
```

Comment:
This has no sense IMO

Reply:
I didn't know if we have units and unit so I started with covering both cases! i can remove it (as I thought in the beginning!

Reply:
Remove it yes :)